### ISSUE TITLE
Link Jump Error (E945: Range too large in character class) with mzlogin/vim-markdown-toc 

### ISSUE THREAD

**Issue by yohannes** [vimwiki/vimwiki#1002]
Error:
I received the following exception due to some kind of conflicts w/  `mzlogin/vim-markdown-toc`, a vim plugin that creates table of contents from markdown headers, as I updated  vimwiki to v2.5:

```
Error detected while processing function vimwiki#base#follow_link[69]..vimwiki#base#open_link[34]..vimw
iki#base#edit_file[32]..<SNR>46_jump_to_anchor[12]..<SNR>46_jump_to_segment:
line   23:
E945: Range too large in character class
```

Steps to Reproduce (Given Vimwiki & vim-markdown-toc are installed already):
1. On a markdown file that has various headers and contents, make sure the cursor is on beginning of the file. Type: `GenTocGFM`  . A table of contents will be generated based on the markdown headers below. 

2. Hit tabs or shift tabs to jump up / down the links. Click enter on any of the links generated by vim-markdown-toc.

3. The error mentioned above will be produce. As a side note, no problem with vimwiki links. 

Any pointer on this issue?

VimwikiShowVersion:
Os: Mac
Vim: 801
Branch: dev
Revision: dbbadc8
Date: 2020-09-01 20:13:12 -0400

mzlogin/vim-markdown-toc is v1.4.0

Vimwiki setting: 
```
let g:vimwiki_list = [{'path': '~/vimwiki/', 'syntax': 'markdown', 'ext': '.md', 'auto_diary_index': 1, 'auto_generate_links': 1, 'auto_toc': 1}]

let g:vimwiki_global_ext = 0

let g:vimwiki_listsyms = '✗○◐●✓'
```

**Comment by tinmarino** [comment]
The error is from here https://github.com/vimwiki/vimwiki/blob/a7f34cd8cfd7e3ab8f1ef3824f5d1233e92f91c6/autoload/vimwiki/base.vim#L867-L872

If you `breakadd func <SNR>46_jump_to_segment>` you may get it. If you send me the link, I may get it

> Click enter on any of the links generated by vim-markdown-toc

Any one ? How do they look like ? Those are converted to regex  and lookued for.

This should not be a very hard issue

If it really is Alll the links, it means that the new `function! vimwiki#base#unnormalize_anchor(anchor) abort` is too premissive on punctuaction: vimwiki#u#get_punctuation_regex(), It may be a mac only issue.

https://github.com/vimwiki/vimwiki/blob/a7f34cd8cfd7e3ab8f1ef3824f5d1233e92f91c6/autoload/vimwiki/base.vim#L747-L821

**Comment by yohannes** [comment]
Hi @tinmarino 

> If you send me the link, I may get it

    Click enter on any of the links generated by vim-markdown-toc

Do you  mean to send you the an example of the auto-generated jump links like what I have below?

```
<!-- vim-markdown-toc GFM -->

+ [Reading](#reading)
+ [Biographies](#biographies)

<!-- vim-markdown-toc -->

# Reading
- A Mind for Numbers: https://www.amazon.com/dp/B00G3L19ZU

# Biographies
- Permanent Record: https://www.amazon.com/dp/1250237238
```

**Comment by tinmarino** [comment]
Yes that is what I meant. Sorry, I have no clue. Those are "easy" headers: no space no special character ....
Try to remove some conf like auto_.. from your vimrc. You can use VimvikiTOC and comment the markdown-toc plugin

**Comment by yohannes** [comment]
It turned out issuing `VimwikiTOC` function on a markdown file that is absent of any toc, also resulted in a similar errors as followed:

```
Error detected while processing function vimwiki#base#table_of_contents[95]..vimwiki#base#update_listing_in_buffer[84]
..677[58]..vimwiki#base#normalize_anchor:
line   21:
E945: Range too large in character class
Error detected while processing function vimwiki#base#table_of_contents[95]..vimwiki#base#update_listing_in_buffer:
line   84:
E714: List require
```

This was the original markdown file:

```
#Contents 
 
# foo

# bar

```

This was the same file after `VimwikiTOC` command was run (w/ the errors aforementioned):

```
#Contents 
 
# foo

# bar

```

Perhaaps I wasn't giving `VimwikiTOC` command the headers format it expects?
